sql/core/src/test/scala/org/apache/spark/sql/RuntimeConfigSuite.scala

```diff
@@ -108,4 +108,26 @@ class RuntimeConfigSuite extends SparkFunSuite {
     // this set should not fail
     conf.set(DEFAULT_PARALLELISM.key, "1")
   }
+
+  test("config entry") {
+    val conf = newConf()
+
+    val entry = SQLConf.FILES_MAX_PARTITION_NUM
+    assert(conf.get(entry.key) === null)
```

Comment:
@HyukjinKwon @cloud-fan this returns null. I was expecting a `NoSuchElementException` here. IMO this is a bug. WDYT?

Reply:
yeah, i think it should. `def get[T](entry: ConfigEntry[T]): T` isn't an API so I think we should be fine with changing this behaviour too.